### ISSUE TITLE
Null suppression doesn't work when the body has a null suppression

### ISSUE THREAD

**Issue by RenderMichael** [nunit/nunit.analyzers#679]
A null suppression throws the null warning suppression out of whack.

One might say why is `str` not getting checked for null. The following is a minimal repro, the actual scenarios I've ran into are less trivial. Suffice it to say, the existence of the null-suppression operator should be evidence enough that it needs to be used sometimes.

``` csharp
internal static string GetString(HasString str)
{
    Assert.That(str.Inner, Is.Not.Null);
    return str.Inner; // works great
}

internal static string GetStringSuppress(HasString? str) // argument is nullable
{
    Assert.That(str!.Inner, Is.Not.Null);
    return str.Inner; // warning: possible null reference return
}

public record HasString(string? Inner);
```

**Comment by manfred-brands** [comment]
In this case `Assert.That(str?.Inner, Is.Not.Null);` would have been a safer statement as it asserts both _str_ and _str.Inner_.

But as you said, your real code is more complicated.

The null-suppression-operator indicates that the writer knows more about the code than the compiler can deduct.
I created a PR to ignore '!' when matching diagnostics with expressions tested with `Is.Not.Null`
